### ISSUE TITLE
Sever crash when placing complex circuit on circuit

### ISSUE THREAD

**Issue by Crazy-KSK** [amadornes/SCM-Localization-Issues#98]
I made [this](https://gist.github.com/e69acb8d174ccd096abc6ccda9beebd8) silly thing to see if something breaks and trying to place it on another circuit does indeed crash the server (singleplayer) with this [error](https://gist.github.com/4092f01ed04dc364be268a01a6dcdd4b).

**Comment by gDKdev** [comment]
Got the same error. Nice silly thing, i think its because we have placed to much things on the circuit

**Comment by amadornes** [comment]
Vanilla really doesn't like long NBT tags... Crashing when sending that tag to the client shouldn't happen anymore in 1.0.2 (to be released), but I'm still not sure how to fix this one...

**Comment by gDKdev** [comment]
What about save all circuits in external files like computercraft does (cc folder in save) and just link to it in the nbt tag (id xyz)

**Comment by amadornes** [comment]
I've considered it before, yeah... But I'm not sure how reliable that would be

**Comment by mcenderdragon** [comment]
@Crazy-KSK have you tried to place the circuit or tried to break the circuit when the crash happened?

**Comment by gDKdev** [comment]
For me it was when i tried to place it

**Comment by Crazy-KSK** [comment]
yeah placing does it, also deleting the blueprint or circuit itself caused the same crash

**Comment by convict999** [comment]
1.0.2... placing 7.07 complexity 8 full adders circuit now causes lockup, exceeding byte payload

**Comment by gDKdev** [comment]
howto calculate the complexity?

**Comment by amadornes** [comment]
Click the circuit with a multimeter

**Comment by gDKdev** [comment]
i guess it actually doesn't matter, because it still crashes and doesn't lockup with 1.0.2...
i'm attempting to place a circuit with 4.1000004 on a circuit with 0.33200002 complexity (still the same from my bugreport - [#95](https://github.com/amadornes/SCM-Localization-Issues/issues/95#event-849059841) )

**Comment by Mr-Gelo** [comment]
I get crash when trying to place this circuit https://gist.github.com/562dbb9de8d151442238c5f23242b573
with 1,062 complexity into a blank circuit in singleplayer. Sometimes it freezes the game and I have crash report (http://pastebin.com/MZsXgQnb) and sometimes it just exits with this message: 
![2016-11-07_16 47 37](https://cloud.githubusercontent.com/assets/23279054/20055198/9aff8dc4-a512-11e6-9724-ea8b44cf6682.png)

**Comment by amadornes** [comment]
I'm happy to say this may actually be fixed in the next build \o/

**Comment by amadornes** [comment]
Still needs some work, of course, but I've managed to split all my packets into ones with <30k bytes each and join them together at the receiving end perfectly :D

**Comment by florianL21** [comment]
Not shure if it is the same probelm, but i've created a 8bit memory cell which consists of 8 single memory cells which are a circuit itself. It has a complexity of about 4. When i try to place my 8bit memory on another circuit board minecraft freezes, and ihave to force close java to get it to behave again.

**Comment by amadornes** [comment]
Yes, it's probably caused by this. If not, it's a StackOverflowError, but you'd have to wait for it to fully crash and then send the report

**Comment by florianL21** [comment]
Yes it probably is, it doesn't crash completley but does write an error to the log which is "Payload may not be larger than 32767 bytes". So it's the same as above.

**Comment by amadornes** [comment]
Ah, okay, then it IS the same issue the rest is having :)

**Comment by yutang13** [comment]
Still having the same problem in 1.0.4. I had made a full adder and was joining 8 together when I placed the 8th I crashed to the multiplayer screen with the same payload error.

Now I get one frame of my world before it "crashes"
